[Joshua Tan] iP

joshtyf opened the pull request.

joshtyf committed:
implemented greet, echo, exit

joshtyf committed:
implemented add, list

joshtyf committed:
implemeted mark as done

joshtyf committed:
implemented todos, events, deadlines

joshtyf committed:
Add A-TextUiTesting

joshtyf committed:
Add error handling

joshtyf committed:
Add delete functionality

joshtyf committed:
Add ability to save tasks to task.txt file

Program  will write the list of tasks to a task.txt file after
every operation that edits the task list.

joshtyf committed:
Add ability to parse strings as Date objects

User inputs will now be parsed as Date objects instead of Strings.

joshtyf committed:
Merge branch 'branch-Level-7' into master

joshtyf committed:
Merge branch 'branch-Level-8' into master

joshtyf committed:
Fix exception handling after branch-Level-8 merge

branch-Level-8 included exception handling that
branch-level-7 did not have. This commit adds the exception handling.

joshtyf committed:
Add a new UI class

Add a new UI class to handle all UI screens

joshtyf committed:
Add a new Storage class

Storage class will handle the reading and writing of files.

joshtyf committed:
Cut the ability to write to file from TaskManager

TaskManager was previously handling all the writes to
"task.txt" file. This functionality is now transferred to
Storage.

joshtyf committed:
Add ability to ask for commands in Ui

Ui class will now print out the interface to ask for a command.

joshtyf committed:
Add a new CommandHandler class

CommandHandler will handle the commands.

joshtyf committed:
Add Command classes for handling of commands

joshtyf committed:
Add exit screen UI

joshtyf committed:
Add TodoCommand class

TodoCommand will handle the execution of adding Todos.

joshtyf committed:
Remove Scanner import statement

joshtyf committed:
Add DeadlineCommand class

DeadlineCommand will handle the addition of deadlines

joshtyf committed:
Add exception throwing to execute() in Command

execute() method will now throw DukeExceptions

joshtyf committed:
Add EventCommand class

EventCommand will handle the addition of events.

joshtyf committed:
Add CompleteCommand class

CompleteCommand will handle the command for completing
tasks.

joshtyf committed:
Add DukeException to markDone() method

joshtyf committed:
Add DeleteCommand class

DeleteCommand will handle the command to delete tasks.

joshtyf committed:
Add DukeException to deleteTask method

joshtyf committed:
Add switch cases to CommandHandler

joshtyf committed:
Add task completed message to Ui class

Task completed message was previously in the Task class.
It is now shifted into the Ui class.

joshtyf committed:
Fix error where task manager cannot display tasks

Storage was previously saving the text contents for Deadline and Events
in the wrong formats, hence they could not be loaded properly.
Format has now been fixed and the issue is resolved.

joshtyf committed:
Cut printing from TaskManager

TaskManagr was previously printing out the list of tasks in its
listTasks() method.
It now returns a string which will be used to print through the Ui class.

joshtyf committed:
Add gradle to the project

joshtyf committed:
Add mavenCentral repository to gradle

joshtyf committed:
Add JUnit tests

joshtyf committed:
Bump gradle shadow plugin to v6.0.0

joshtyf committed:
Cut gradle application plugin

Plugin was causing issues when running shadowJar task.
The main problem was that the mainClass was not defined for application
and shadowJar was not able to run properly.

joshtyf committed:
Add shadowJar manifest config

joshtyf committed:
Document Command.java

joshtyf committed:
Document CommandHandler.java

joshtyf committed:
Document CompleteCommand.java

joshtyf committed:
Document Deadline.java

joshtyf committed:
Document DeadlineCommand.java

joshtyf committed:
Document DeleteCommand.java

joshtyf committed:
Document Duke.java

joshtyf committed:
Document DukeException.java

joshtyf committed:
Document ErrorCommand.java

joshtyf committed:
Document Event.java

joshtyf committed:
Document EventCommand.java

joshtyf committed:
Document ExitCommand.java

joshtyf committed:
Document ListCommand.java

joshtyf committed:
Document Storage.java

joshtyf committed:
Document Task.java

joshtyf committed:
Document TaskManager.java

joshtyf committed:
Document Todo.java

joshtyf committed:
Document TodoCommand.java

joshtyf committed:
Document Ui.java

joshtyf committed:
Refactor CommandHandler.java

joshtyf committed:
Refactor Duke.java

joshtyf committed:
Refactor Storage.java

joshtyf committed:
Refactor Task.java

joshtyf committed:
Refactor TaskManager.java

joshtyf committed:
Add findTask method to TaskManager.java

TaskManager will now be able to support searching for a task
given a keyword

joshtyf committed:
Add find operation

joshtyf committed:
Add FindCommand.java

FindCommand is a new command that will be used to search for tasks with
a particular keyword.

joshtyf committed:
Add askForKeyword method

joshtyf committed:
Merge branch 'branch-A-JavaDoc' into master

joshtyf committed:
Merge branch 'branch-A-CodingStandard' into master

joshtyf committed:
Merge branch 'branch-Level-9' into master

joshtyf committed:
Refactor into commands package

joshtyf committed:
Refactor tasks

joshtyf committed:
Refactor and create duke package

joshtyf committed:
Refactor commands and tasks package into duke

joshtyf committed:
Make TaskManager public

joshtyf committed:
Make Task public

joshtyf committed:
Make DukeException public

joshtyf committed:
Add import statements

joshtyf committed:
Change tasks.txt file path

joshtyf committed:
Add import statements

joshtyf committed:
Add import statements

joshtyf committed:
Fix ui statements

joshtyf committed:
Fix print statements

joshtyf committed:
Merge branch 'branch-A-Packages' into master

joshtyf committed:
Copy config from ip repo

joshtyf committed:
Merge branch 'branch-A-Gradle' into master

joshtyf committed:
Add JavaFx dependency

joshtyf committed:
Add Launcher.java

Create a Launcher class as per instructions from the guide.

joshtyf committed:
Add code to start the JavaFx GUI

joshtyf committed:
Add more elements to GUI

joshtyf committed:
Add custom size to GUI

joshtyf committed:
Add user input handling

joshtyf committed:
Add vertical scrolling

joshtyf committed:
Merge branch 'branch-Level-10' into master

wenhaogoh commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,286 @@
+import javafx.scene.layout.Region;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+import java.util.Scanner;
```
Should the java imports be at the top? I noticed the same issues in several other places too.

wenhaogoh commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,88 @@
+ * <code>duke.Storage</code> handles all file read and writes.
+ */
+public class Storage {
+    final static String TASKS_TEXT_FILE_PATH = "../../tasks.txt";
```
Should there be an access modifier? I noticed the same issue in several other places too.

joshtyf replied:
You are right! I forgot about them. Thanks for highlighting it.

wenhaogoh commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,88 @@
+                String taskInfo = sc.nextLine();
+                String[] arr = taskInfo.split(",");
+                switch (arr[0]) {
+                    case "T":
```
Should the case block have 4 less indentation spaces? I noticed this issue in several other places too.

joshtyf replied:
Oops, I forgot to change the settings on my IDE. Thanks for highlighting the issue.

wenhaogoh reviewed (commented):
The code is easy to read. Just some nits to fix. Good work!

joshtyf committed:
Add profile pictures

joshtyf committed:
Add DialogBox.java class

joshtyf committed:
Add fxml files

joshtyf committed:
Add Main.java class for loading GUI

joshtyf committed:
Add MainWindow controller

joshtyf committed:
Make app to launch from Main

joshtyf committed:
Refactor GUI to Main and MainWindow

joshtyf committed:
Add DialogBox.java

joshtyf committed:
Merge branch 'branch-Level-10' into master

joshtyf committed:
Make constructor public

joshtyf committed:
Remove dead code

joshtyf committed:
Remove useless import statements

joshtyf committed:
Integrate with GUI

joshtyf committed:
Fix wrong tasks.txt file path

The file path for tasks.txt was previously defined wrongly.
It has now been fixed.

joshtyf committed:
Make use of super constructor for error msg

joshtyf committed:
Integrate with GUI

joshtyf committed:
Document Command.java

joshtyf committed:
Document CommandHandler.java

joshtyf committed:
Reformat switch indentations

joshtyf committed:
Document CompleteCommand.java

joshtyf committed:
Document DeadlineCommand.java

joshtyf committed:
Reformat switch indentation

joshtyf committed:
Document DeleteCommand.java

joshtyf committed:
Fix call to wrong method

joshtyf committed:
Document DeleteCommand.java

joshtyf committed:
Document ErrorCommand.java

joshtyf committed:
Document Command.java

joshtyf committed:
Document CompleteCommand.java

joshtyf committed:
Document Deadline.java

joshtyf committed:
Document DeleteCommand.java

joshtyf committed:
Document EventCommand.java

joshtyf committed:
Reformat indentations

joshtyf committed:
Document ExitCommand.java

joshtyf committed:
Document FindCommand.java

joshtyf committed:
Document ListCommand.java

joshtyf committed:
Document TodoCommand.java

joshtyf committed:
Document ListCommand.java

joshtyf committed:
Document FindCommand.java

joshtyf committed:
Document Deadline.java

joshtyf committed:
Document Event.java

joshtyf committed:
Document Task.java

joshtyf committed:
Document TaskManager.java

joshtyf committed:
Document Todo.java

joshtyf committed:
Refactor stringbuffer appends

joshtyf committed:
Document DialogBox.java

joshtyf committed:
Reformat and remove dead code

joshtyf committed:
Rename setNewCommand to setCommand

joshtyf committed:
Document Duke.java

joshtyf committed:
Document Main.java

joshtyf committed:
Document Storage.java

joshtyf committed:
Document Ui.java

joshtyf committed:
Merge branch 'branch-Level-10' into master

joshtyf committed:
Add assertions

joshtyf committed:
Add exception handling

joshtyf committed:
Remove dead code

joshtyf committed:
Document Ui.java

joshtyf committed:
Refactor switch statements

joshtyf committed:
Merge branch 'master' into branch-A-Assertions

joshtyf committed:
Merge pull request #1 from joshtyf/branch-A-Assertions

Add assertions

joshtyf committed:
Merge branch 'master' into branch-A-CodeQuality

joshtyf committed:
Merge pull request #2 from joshtyf/branch-A-CodeQuality

Branch A-CodeQuality

joshtyf committed:
Merge branch 'master' into branch-A-CodeQuality

joshtyf committed:
Fix missing throws declaration

joshtyf committed:
Change add task method

joshtyf committed:
Implement Comparable interface

joshtyf committed:
Use Text widget instead of Label

Using a Label widget will result in the truncation of the duke
responses. Using a Text widget prevents that.

joshtyf committed:
Change loading procedure from tasks.txt

joshtyf committed:
Add sorting functionality

Deadline and Event classes now implement Comparable interface and will
sort according to their start times.

joshtyf committed:
Refactor exception handling for invalid index

joshtyf committed:
Add NumberFormatException handling

joshtyf committed:
Reformat exception message

Add a new line before showing the exception message

joshtyf committed:
Document Deadline.java

joshtyf committed:
Document TaskManager.java

joshtyf committed:
Document Event.java

joshtyf committed:
Document Storage.java

joshtyf committed:
Merge branch 'branch-Sorting' into master

joshtyf committed:
Refactor for loops and use streams instead

joshtyf committed:
Merge pull request #3 from joshtyf/branch-A-Streams

Refactor for loops and use streams instead

joshtyf committed:
Remove unuse import

joshtyf committed:
Reformat wildcard import to single class imports

yanbingtao commented on src/main/java/duke/tasks/TaskManager.java:
```diff
@@ -0,0 +1,222 @@
+    public String listTasks() {
+        int i = 1;
+        StringBuilder sb = new StringBuilder();
+        if (!todos.isEmpty()) {
```
you may consider using guard clauses technique here? So that you could reduce the depth of your code (:

yanbingtao commented on src/main/java/duke/tasks/TaskManager.java:
```diff
@@ -0,0 +1,222 @@
+        }
+        if (!deadlines.isEmpty()) {
+            sb.append("Here are your deadlines:\n");
+            for (Deadline deadline : deadlines) {
```
This for loop is quite similar with the for loop in line 126, you may consider extracting out as a method? (:

joshtyf committed:
Improve ui of the dialog box

joshtyf committed:
Remove old picture

joshtyf committed:
Add new picture

joshtyf committed:
Use new pictures

joshtyf committed:
Improve ui of main window

joshtyf committed:
Merge branch 'branch-Better-GUI' into master

joshtyf committed:
Merge pull request #4 from joshtyf/branch-A-CodeQuality

Merge branch 'master' into branch-A-CodeQuality

joshtyf committed:
Fix indexing bug

joshtyf committed:
Add ui screenshot

joshtyf committed:
Add ui images

joshtyf committed:
Update user guide

joshtyf committed:
Merge branch 'master' of https://github.com/joshtyf/ip into master

joshtyf committed:
Reduce image size

joshtyf committed:
Update header links

joshtyf committed:
Reduce image size

joshtyf committed:
Remove random line

joshtyf committed:
Improve main window gui

joshtyf committed:
Improve image quality

joshtyf committed:
Add exit functionality

joshtyf committed:
Change image resolution

joshtyf committed:
Change image resolution

Previous commit increased the resolution. This commit decreases it.

joshtyf committed:
Reformat newline

joshtyf committed:
Update README.md

joshtyf committed:
Add ui screenshot

joshtyf committed:
Refactor variable name

Changed the variable name from cmd to newCmd as it might be confused with the old command variable which was also called cmd.

joshtyf committed:
Add CancelCommand.java

joshtyf committed:
Add cancel command functionality

joshtyf committed:
Reformat command inputs

"/" to distinguish commands from normal inputs

joshtyf committed:
Add cancel command

joshtyf committed:
Remove TODO

joshtyf committed:
Reformat response string

Improve error response information

joshtyf committed:
Add isEmpty method

joshtyf committed:
Make init method check if the task list is empty

joshtyf committed:
Remove TODO

joshtyf committed:
Make execute method return true

execute method previously returned false but now returns true

joshtyf committed:
Document Command.java

execute method will now always return true

joshtyf committed:
Add System.exit

joshtyf committed:
Replace Ui.png

joshtyf committed:
Replace ui screenshots

joshtyf committed:
Update README.md

joshtyf committed:
Add new ui screenshots

joshtyf committed:
Update build.gradle

joshtyf committed:
Update file paths
